src/discordtipbot/discordbot.cr

```diff
@@ -381,17 +376,27 @@ class DiscordBot
     when false
       reply(msg, "**ERROR**: There was a problem trying to transfer funds. Please try again later. If the problem persists, please contact the dev for help in #{@config.prefix}support")
     when true
-      string = ""
       amount_each = amount / authors.size
 
-      if @tip.get_config(guild_id(msg), "mention")
-        authors.each { |x| string = string + ", <@#{x}>" }
+      string = build_user_string(get_config_mention(msg), authors)
+
+      reply(msg, "**#{msg.author.username}** rained a total of **#{amount} #{@config.coinname_short}** (#{amount_each} #{@config.coinname_short} each) onto #{string}")
+    end
+  end
+
+  private def build_user_string(mention : Bool, users : Array(UInt64))
+    string = String.build do |str|
+      if mention
+        users.each { |x| str << ", <@#{x}>" }
       else
-        authors.each { |x| string = string + ", #{@cache.resolve_user(x).username}" }
+        users.each { |x| str << ", #{@cache.resolve_user(x).username}" }
       end
-      string = string.lchop(", ")
-      reply(msg, "**#{msg.author.username}** rained a total of **#{amount} #{@config.coinname_short}** (#{amount_each} #{@config.coinname_short} each) onto #{string}")
     end
+    string = string.lchop(", ")
+  end
+
+  private def get_config_mention(msg)
+    @tip.get_config(guild_id(msg), "mention") || false
```

Comment:
I think it would be cleaner to set defaults in `get_config` so that it returns `Bool` instead of `Bool | Nil`. Maybe fit for another PR though.

Reply:
Added a note to my TODO list